driver-core/src/main/java/com/datastax/driver/core/AggregateMetadata.java

```diff
@@ -101,11 +106,29 @@ static AggregateMetadata build(KeyspaceMetadata ksm, Row row, VersionNumber vers
         if (version.getMajor() >= 3) {
             stateType = DataTypeCqlNameParser.parse(stateTypeName, cluster, ksm.getName(), ksm.userTypes, null, false, false);
             String rawInitCond = row.getString("initcond");
-            initCond = rawInitCond == null ? null : codecRegistry.codecFor(stateType).parse(rawInitCond);
+            if (rawInitCond == null) {
+                initCond = null;
+            } else {
+                try {
+                    initCond = codecRegistry.codecFor(stateType).parse(rawInitCond);
```

Comment:
It looks like this actually succeeds in the case the test case was created for, so the exception isn't raised and `rawInitCond` isn't used..  `initCond` is set to a `ByteBuffer` (so `getInitCond()` also returns a `ByteBuffer`).  It looks like the test is expecting it to return a string literal (i.e.: `'s@foo:i@32'`).  The cql string for the aggregate looks like:

> CREATE AGGREGATE ks_1.ag() SFUNC id STYPE 'org.apache.cassandra.db.marshal.DynamicCompositeType(s=>org.apache.cassandra.db.marshal.UTF8Type,i=>org.apache.cassandra.db.marshal.Int32Type)' INITCOND 0x80730003666f6f00806900040000002000;

Is this the expected behavior?